### DISCUSSION TITLE
[Question] EfficientAD is patented by MVTec in Japan

### DISCUSSION THREAD

**Discussion by SachihiroNakagawa** [open-edge-platform/anomalib#2526] [Q&A] [answered]
It seems that EfficientAD is patented by MVTec, at least in Japan.
Is there any issue with the license if I incorporate EfficientAD from Anomalib into commercial software?

https://patentscope2.wipo.int/search/en/detail.jsf?docId=JP394418688&_fid=EP393389056

**Comment by samet-akcay** [comment]
Are you sure if it is Efficient AD? I cannot see the details due to the language barrier. However, when I look at the list of the authors of the patent documents, I can see they don't match the that of Efficient AD authors. Paul Bergman, for example, is the author of the dataset, some previous mvtec papers, but he is currently not with mvtec anymore.

**Comment by samet-akcay** [comment]
Also, the EfficientAD in Anomalib was implemented by @alexriedel1, by following the details in the paper, which is publicly available on ArXiv.

**Comment by SachihiroNakagawa** [comment]
@samet-akcay 
Thank you for your response.
I'm not a patent expert, so I need more research and confirmation.

**Comment by alexriedel1** [comment]
It is also patented in europe under this patent. It didnt fully read it yet and dont know if it would affect EfficientAD
 https://worldwide.espacenet.com/publicationDetails/originalDocument?CC=EP&FT=D&KC=A1&NR=4145401A1&flavour=trueFull&locale=en_EP&popup=true

**Comment by samet-akcay** [comment]
Thanks for sharing @alexriedel1. I could contact the authors to see

**Comment by suahelen** [comment, marked as answer]
Quick update on the issue. I had contact with the authors of the paper.

**Comment by MVTec Software GmbH:** 
_In general, GCAD as well as EfficientAD can fall within the scope of the patent. Consequently, we strongly discourage the use of open-source implementations of the algorithms, e.g. provided within the anomalib library, to avoid IP conflicts and legal consequences._

So for now I would not recommend to further make use of this architecture unless used for small private projects

**Comment by samet-akcay** [comment]
Our plan is to implement [GLASS](https://arxiv.org/abs/2407.09359v1) and [GLAD](https://arxiv.org/abs/2406.07487v3) papers into Anomalib, both of which already outperforms EfficientAd. Once added, these models could be used instead of EfficientAd.

**Reply by lekklev** [reply]
That's great news. Where can we track development of the implemention of these models?

**Comment by suahelen** [comment]
That some lovely news. 
I have played around with glass a bit and it seems to work promising. Looking forward to that.
